Sources/Private/Experimental/Layers/MaskCompositionLayer.swift

```diff
@@ -0,0 +1,104 @@
+// Created by Cal Stephens on 1/6/22.
+// Copyright © 2022 Airbnb Inc. All rights reserved.
+
+import QuartzCore
+
+// MARK: - MaskCompositionLayer
+
+/// The CALayer type responsible for rendering the `Mask` of a `BaseCompositionLayer`
+final class MaskCompositionLayer: CALayer {
+
+  // MARK: Lifecycle
+
+  init(masks: [Mask]) {
+    maskLayers = masks.map { MaskLayer(mask: $0) }
+    super.init()
+
+    for maskLayer in maskLayers {
+      addSublayer(maskLayer)
+    }
+  }
+
+  required init?(coder _: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  /// Called by CoreAnimation to create a shadow copy of this layer
+  /// More details: https://developer.apple.com/documentation/quartzcore/calayer/1410842-init
+  override init(layer: Any) {
+    guard let layer = layer as? Self else {
+      fatalError("\(Self.self).init(layer:) incorrectly called with \(type(of: layer))")
+    }
+
+    maskLayers = layer.maskLayers
+    super.init(layer: layer)
+  }
+
+  // MARK: Internal
+
+  override func layoutSublayers() {
+    super.layoutSublayers()
+
+    for sublayer in sublayers ?? [] {
+      sublayer.fillBoundsOfSuperlayer()
+    }
+  }
+
+  // MARK: Private
+
+  private let maskLayers: [MaskLayer]
+
+}
+
+// MARK: AnimationLayer
+
+extension MaskCompositionLayer: AnimationLayer {
+  func setupAnimations(context: LayerAnimationContext) {
+    for maskLayer in maskLayers {
+      maskLayer.setupAnimations(context: context)
+    }
+  }
+}
+
+// MARK: - MaskLayer
+
+extension MaskCompositionLayer {
+  final class MaskLayer: CAShapeLayer {
+
+    // MARK: Lifecycle
+
+    init(mask: Mask) {
+      maskModel = mask
+      super.init()
+      fillColor = "#000000".cgColor
```

Comment:
nit: why not `UIColor.black.cgColor`? Seems like we'll pay for string processing overhead here
```suggestion
      fillColor = UIColor.black.cgColor
```

Reply:
`CGColor.black` is only defined on macOS :(

Reply:
I'll tweak this to not use a string though, good call

Reply:
Ah interesting! Yes then perhaps `UIColor.black.cgColor`

Reply:
We can't import `UIKit` / `UIColor` here because this file is also used on macOS (non-Catalyst) 

`CGColor` initializers are truly a wild-west, there's a nice `CGColor(red:green:blue:alpha:)` initializer that was added in iOS 13 but we're using a minimum deployment target of iOS 11. 

The pre-iOS 13 initializer isn't too bad, I just don't love how you have to force unwrap it 🤷🏻 

```swift
CGColor(
  colorSpace: CGColorSpaceCreateDeviceRGB(),
  components: [0, 0, 0])!
```

may write a "polyfill" init like we're currently doing in the `String.cgColor` extension:

```swift
    if #available(iOS 13.0, tvOS 13.0, *) {
      return CGColor(red: red, green: green, blue: blue, alpha: 1)
    } else {
      return CGColor(
        colorSpace: CGColorSpaceCreateDeviceRGB(),
        components: [red, green, blue])!
    }
```

Reply:
i cry 😆 😥

<img width="904" alt="image" src="https://user-images.githubusercontent.com/1811727/148594559-3b70c88f-46b0-4e7e-a9f6-fbab2f437d6b.png">

Reply:
Fixed: https://github.com/airbnb/lottie-ios/pull/1468/commits/c981d5e601fefc76bdca25bdac784963db3d26dd